cellprofiler/modules/savecroppedobjects.py

```diff
@@ -255,11 +255,21 @@ def run(self, workspace):
             workspace.display_data.filenames = filenames
 
     def upgrade_settings(self, setting_values, variable_revision_number, module_name):
-        if variable_revision_number < 3:
+        if variable_revision_number == 2:
             # Older module version, revert to not using file names in output crops
-            # Also, reorder setting_values
+            # Also, reorder setting_values to reflect order of settings in the GUI. Original order:
+            # [objects_name, directory, file_format, export_option, image_name]
+            # New order:
+            # [export_option, objects_name, directory, use_filename, file_image_name, nested_save, file_format, image_name]
             setting_values = (
-                [setting_values[3]] + setting_values[:2] + [False, "No", None] + [setting_values[2]] + [setting_values[4]]
+                [setting_values[3]] + setting_values[:2] + [False, "None", False] + [setting_values[2]] + [setting_values[4]]
+            )
+            variable_revision_number = 3
+
+        if variable_revision_number == 1:
```

Comment:
This needs to be first, so that if someone has a 1, it bumps to a 2, and then at the end of that if statement it will be a 2, so now can bump to a 3, etc

Reply:
Ah, nvm, I see what you did - for parallelness, if you don't mind following that structure (1 bumps to 2 bumps to 3 etc as opposed to a "2 OR 1 bumps to 3, then 3 bumps to 4, etc"), I think future-us-es will be less confused.